app/controllers/videos_controller.rb

```diff
@@ -7,9 +7,22 @@ def index
 
   def show
     @video = Video.find(params[:id])
+    @comment = Comment.new
   end
 
   def search
     @search_result = Video.search_by_title(params[:title])
   end
+
+  def comment
+    @video = Video.find(params[:id])
+    @comment = @video.comments.build(rate: params[:comment][:rate], user: current_user, review: params[:comment][:review])
+
+    if @comment.save
+      flash[:info] = "new comment added!"
+      redirect_to video_path
+    else
+      render :show
```

Comment:
If comment is valid, I like to display flash message, if invalid,  I want to display error message above the comment section and error hint on the fields.